### ISSUE TITLE
Getting "fatal error: 'math.h' file not found" on OS X 10.11

### ISSUE THREAD

**Issue by michaelochurch** [commercialhaskell/stack#1291]
I recently upgraded my MacBook to OS X 10.11 and installed Stack using ```brew install haskell-stack```. 

When I tried to run ```stack ghci``` in a brand-new project, I got this error (which is possibly not stack's fault):

```
$ stack ghci
[1 of 1] Compiling Main             ( /private/var/folders/k5/2lpm427x5bqgmgybtpyvv4qm0000gn/T/stack1383/Setup.hs, /private/var/folders/k5/2lpm427x5bqgmgybtpyvv4qm0000gn/T/stack1383/Setup.o )
Linking /Users/michaelchurch/.stack/setup-exe-cache/tmp-setup-Simple-Cabal-1.22.4.0-x86_64-osx-ghc-7.10.2 ...

In file included from /var/folders/k5/2lpm427x5bqgmgybtpyvv4qm0000gn/T/ghc1397_0/ghc_4.c:1:0:


In file included from /usr/local/Cellar/ghc/7.10.2/lib/ghc-7.10.2/include/Rts.h:30:0:


/usr/local/Cellar/ghc/7.10.2/lib/ghc-7.10.2/include/Stg.h:69:10:
     fatal error: 'math.h' file not found
#include <math.h>
         ^
1 error generated.
Exit code ExitFailure 1 while running ["ghc","-clear-package-db","-global-package-db","-hide-all-packages","-package","base","-package","Cabal-1.22.4.0","/private/var/folders/k5/2lpm427x5bqgmgybtpyvv4qm0000gn/T/stack1383/Setup.hs","-o","/Users/michaelchurch/.stack/setup-exe-cache/tmp-setup-Simple-Cabal-1.22.4.0-x86_64-osx-ghc-7.10.2"] in /private/var/folders/k5/2lpm427x5bqgmgybtpyvv4qm0000gn/T/stack1383/
```

I don't expect this to be difficult to work around, and it could be something that went wrong with the OS X upgrade, but I'd imagine that this sort of thing would be daunting to newcomers.

**Comment by mgsloan** [comment]
Thanks for the report!  Potentially related to #1279, but that's on windows.

**Comment by borsboom** [comment]
I just tried to reproduce this following these steps, but didn't run into the error:

1. Upgrade to El Capitan (from Yosemite)
2. Install Homebrew (this machine had never had Homebrew on it before)
    a. As part of this, the Xcode and the Xcode CLI tools were installed
3. Install Stack using `brew install haskell-stack`
4. Create a project using `stack new`
5. Run `stack ghci` in the new project directory

Did you install Xcode after upgrading?  I've noticed every OS upgrade requires re-installing it.  I think `xcode-select install` will do that.
